assets/src/edit-story/elements/image/display.js

```diff
@@ -47,12 +69,28 @@ function ImageDisplay({ element, box }) {
     focalY
   );
 
+  useEffect(() => {
+    let timeout;
+    if (window.webStoriesEditorResourceList[resource.id]?.type !== 'fullsize') {
```

Comment:
This hook doesn't re-trigger based on this external value. Is that a problem or a feature?

Reply:
Because the Image element is re-rendered many times (initially and if you do anything in the editor) we want to start loading the full-size image just once (and once per resource.id) and avoid any extra state updates. The flow here is not straightforward but should work in practice.